java/gateway-core/src/main/java/com/avaulta/gateway/pseudonyms/PseudonymImplementation.java

```diff
@@ -25,4 +29,13 @@ public static PseudonymImplementation parseHttpHeaderValue(String httpHeaderValu
         }
         throw new IllegalArgumentException("Unknown pseudonym implementation: " + httpHeaderValue);
     }
+
+    public static PseudonymImplementation parseConfigPropertyValue(String configPropertyValue) {
+        for (PseudonymImplementation impl : values()) {
+            if (impl.getHttpHeaderValue().equals(configPropertyValue)) {
+                return impl;
+            }
+        }
+        throw new IllegalArgumentException("Unknown pseudonym implementation: " + configPropertyValue);
+    }
```

Comment:
methods are identical, maybe merge both into `parseValue` ??

Reply:
A functional approach
```suggestion
    public static PseudonymImplementation parseValue(String configPropertyValue) {
        return Arrays.stream(values())
            .filter( p -> p.getHttpHeaderValue().equals(httpHeaderValue))
            .findFirst()
            .orElseThrow( () -> new IllegalArgumentException("Unknown pseudonym implementation: " + httpHeaderValue));
    }
```